test/common/testlib.py

```diff
@@ -743,27 +761,30 @@ def setUp(self):
         if opts.address and self.provision is not None:
             raise unittest.SkipTest("Cannot provision multiple machines if a specific machine address is specified")
 
-        self.machine = None
         self.browser = None
         self.machines = {}
         provision = self.provision or {'machine1': {}}
```

Comment:
Please move that into the `else:` branch as well, to make sure we don't use it in the nondestructive case.